examples/encoders/encoder_example.py

```diff
@@ -1,19 +1,134 @@
+  features = featurizer.featurize(mols)
+
+  dataset = NumpyDataset(features, features)
+  prediction = tf_enc.predict_on_batch(dataset.X)
+
+  tf_de = TensorflowMoleculeDecoder.zinc_decoder()
+  one_hot_decoded = tf_de.predict_on_batch(prediction)
+  print(featurizer.untransform(one_hot_decoded))
+
+
+class LossLayer(Layer):
+
+  def __init__(self, **kwargs):
+    super().__init__(**kwargs)
+
+  def set_parents(self, parents):
+    z_mean, z_std, input_layer, decoded = parents[0], parents[1], parents[
+        2], parents[3]
+    latent_loss = 0.5 * tf.reduce_sum(tf.square(z_mean.out_tensor) + \
+                                      tf.square(z_std.out_tensor) - \
+                                      tf.log(tf.square(z_std.out_tensor)) - 1, 1)
+    generation_loss = tf.reduce_mean(
+        tf.nn.softmax_cross_entropy_with_logits(
+            logits=decoded.out_tensor, labels=input_layer.out_tensor))
+    self.out_tensor = tf.reduce_mean(generation_loss + latent_loss)
+
+
+def main2():
+  from deepchem.feat.one_hot import zinc_charset
+  dense_hidden_size = 435
+  latent_vector_size = 292
+  gru_hidden = 100
+  gru_channels = 501
+  batch_size = 200
+
+  # import json
+  #
+  # smiles = json.load(open("/home/leswing/Documents/keras-molecules/out.json"))
+  # mols = []
+  # for smile in smiles:
+  #   smile = "".join(smile)
+  #   try:
+  #     mol = Chem.MolFromSmiles(smile)
+  #     if mol is None:
+  #       continue
+  #     mols.append(mol)
+  #   except Exception as e:
+  #     print(e, smile)
+  #
+  # featurizer = OneHotFeaturizer()
+  # features = featurizer.featurize(mols)
+  #
+  # dataset = NumpyDataset(features, features)
+  dataset = DiskDataset(data_dir='/tmp/tmpza_f9bd_')
+  print(dataset.X.shape)
+
+  graph_model = TensorGraph(
+      data_dir="/home/leswing/Documents/graph_model/graph_model")
+  input_layer = Input((batch_size, 120, 47), name="features")
+  conv1 = Conv1DLayer(width=9, out_channels=9, name="conv1")
+  conv2 = Conv1DLayer(width=9, out_channels=9, name="conv2")
+  conv3 = Conv1DLayer(width=10, out_channels=11, name="conv3")
+  flatten = Flatten(name="flatten")
+  dense1 = Dense(out_channels=dense_hidden_size, name="dense1")
+  z_mean = Dense(out_channels=latent_vector_size, name="z_mean")
+  z_std = Dense(out_channels=latent_vector_size, name="z_std")
+  latent = CombineMeanStd(name="latent")
+  dense2 = Dense(out_channels=latent_vector_size, name="dense2")
+  repeat1 = Repeat(n_times=120)
+  gru1 = GRU(n_hidden=gru_hidden,
+             out_channels=gru_channels,
+             batch_size=batch_size,
+             name="GRU1")
+  gru2 = GRU(n_hidden=gru_hidden,
+             out_channels=gru_channels,
+             batch_size=batch_size,
+             name="GRU2")
+  gru3 = GRU(n_hidden=gru_hidden,
+             out_channels=gru_channels,
+             batch_size=batch_size,
+             name="GRU3")
+  ts_dense = TimeSeriesDense(out_channels=47)
+
+  graph_model.add_layer(input_layer, parents=list())
```

Comment:
add_layer already adds the additional layer -- I thought it was cleaner if we didn't use it in the example :).  I actually use that syntax for the loss layer.